### PR TITLE
done task-7

### PR DESCRIPTION
**Pull request by demasha0908**
https://demasha0908.github.io/fe-react-2024/

- [x] Mentor is added to PR
- [x]  AC copied to PR description
- [x]  Link to deployed version is added to PR
- [x]  ProductPage fetches data from the API using either fetch or axios
- [x]  Pagination is implemented using query parameters
- [x]  Search functionality is implemented using query parameters, with the search triggered by a search button
- [x]  Category filtering is implemented using query parameters, resetting the page to the first page on change
- [x]  A loading spinner or message is displayed while awaiting API requests
- [x]  If the API response is an empty array or an error occurs, a "No products" message is displayed instead of a blank screen

Acceptance Criteria for 1*

- [x]  Pagination is hidden on mobile view
- [x]  Infinite scroll is implemented for mobile view, either manually or using a library
- [ ]  Error notifications are implemented using axios interceptors, with component libraries for toastr notifications

### CODE REVIEW

File: src/components/product/ProductsPage.component.tsx [hunk 1]
```diff
@@ -1,49 +1,87 @@
...
+        setProducts([]);
     };
 
+    const handleScroll = useCallback(() => {
+        if (
+            window.innerHeight + document.documentElement.scrollTop >= document.documentElement.offsetHeight - SCROLL_THRESHOLD &&
+            !isLoading &&
+            currentPage < totalPages
+        ) {
+            setCurrentPage(currentPage + 1);
```
Comment (vladyslav-yermolin-moc) [inline src/components/product/ProductsPage.component.tsx:67]: Hey, @demasha0908, when this condition is met and a new bunch of products arrives, the scroll jumps to the page top. Please avoid this jumping.

File: src/components/product/ProductCard.component.tsx [hunk 2]
```diff
@@ -9,12 +14,26 @@ interface ProductCardProps {
 }
 
 const ProductCard: React.FC<ProductCardProps> = ({ product }) => {
+    const { isError, isLoading } = useContext(ProductsDataContext);
+
     const navigate = useNavigate();
 
     const showProductPage = () => {
```
Comment (vladyslav-yermolin-moc) [inline src/components/product/ProductCard.component.tsx:21]: If you don't have a product page, proly you should remove it.
image

File: src/App.tsx [hunk 3]
```diff
@@ -4,15 +4,18 @@ import AboutPage from '@/components/about/About.component.tsx';
...
-        <Routes>
-            <Route path="/" element={<LayoutComponent />}>
-                <Route index element={<AboutPage />} />
-                <Route path="products" element={<ProductsPage />} />
-                <Route path="*" element={<NotFound />} />
-            </Route>
-        </Routes>
+        <ProductsDataContextProvider>
+            <Routes>
+                <Route path="/" element={<LayoutComponent />}>
```
Comment (vladyslav-yermolin-moc) [inline src/App.tsx:13]: Please add a link to about page to the site logo
image

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]

File: src/components/product/ProductsPage.component.tsx (hunk 1 above)
Comment (demasha0908) [reply src/components/product/ProductsPage.component.tsx:67]: Fixed

File: src/components/product/ProductCard.component.tsx (hunk 2 above)
Comment (demasha0908) [reply src/components/product/ProductCard.component.tsx]: Fixed

File: src/App.tsx (hunk 3 above)
Comment (demasha0908) [reply src/App.tsx:13]: Fixed

File: src/components/product/ProductsPage.component.tsx (hunk 1 above)
Comment (vladyslav-yermolin-moc) [reply src/components/product/ProductsPage.component.tsx:67]: I still see this jump, send you a screen recording in telegram
Resolve this issue, for now as it is not a critical one.

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Pull request merged**